src/container/srv_target.c

```diff
@@ -752,7 +752,14 @@ ds_cont_child_cache_destroy(struct daos_lru_cache *cache)
 static void
 cont_child_put(struct daos_lru_cache *cache, struct ds_cont_child *cont)
 {
+	bool		 wake_cond = false;
+
+	if (cont->sc_stopping == true && daos_lru_ref_count(&cont->sc_list) == 3)
```

Comment:
Because in cont_child_destroy_one() it hold one extra refrences, and conf ref is inited as 1. so if ref count is 2 that means there is no other references for this containers.

Logic is like firstly we set sc_stopping for containers, and new lookup will fail.
but we wait existed references finish, that is why we could not wake up as long as sc_stopping is set. makes sense?

Reply:
I think we'd try to avoid introducing such ad-hoc reference checking code further.

I think the '3' means we assume the daos_lru_cache for ds_cont_child is always created without D_HASH_FT_EPHEMERAL (no reference increasing when adding to cache) flag, I think we'd better don't make such assumption.

It looks to me current daos_lru code is flawed: in daos_lru_ref_release() it assumes 'in cache' holds one reference, but daos_lru_cache_create() doesn't regard D_HASH_FT_EPHEMERAL as an invalid feature for daos_lru.

I suggest implementing the common 'waiting for the last reference on daos_lru_ref_release()' for daos_lru, for example: introduce two more callbacks in daos_llink_ops, one for 'wakeup' and the other for 'wait', they could be called in daos_lru_ref_release() accordingly.

BTW, if this ticket isn't urgent, should we use this opportunity try to incorporate some the changes for what we discussed on I/O meeting? Thanks.